File: src/2020/06.rb
groups = File.open("../../data/2020/06.txt").read.split("\n\n").map &:split

part_1 = groups.sum { |g| g.join.chars.uniq.size }

puts part_1

part_2 = groups.sum { |g| g.map(&:chars).inject(:&).size }

puts part_2

File: src/2016/06.py
import pathlib
from collections import Counter

from utils import *

data = pathlib.Path("../../data/2016/06.txt").read_text(encoding="utf-8")
lines = data.strip().split("\n")

columns = transpose([list(line) for line in lines])

counters = [Counter(column).most_common() for column in columns]

part_1 = "".join([c[0][0] for c in counters])
part_2 = "".join([c[-1][0] for c in counters])

print(part_1)
print(part_2)
